### ISSUE TITLE
Add nofollow to Link UI in Nav and Buttons blocks

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#54091]
Following on from https://github.com/WordPress/gutenberg/pull/53945 we should add `Nofollow` control to other instances of the Link UI. Specifically:

- Navigation
- Buttons

------

 If and when this merges we should follow up to add this control elsewhere:

- Navigation links
- Buttons block

_Originally posted by @getdave in https://github.com/WordPress/gutenberg/issues/53945#issuecomment-1700772084_

**Comment by bangank36** [comment]
@getdave can I contribute to the `Button` block link editor with `nofollow` control? Thanks
<img width="640" alt="image" src="https://github.com/WordPress/gutenberg/assets/10071857/443e747f-2867-4513-93ee-dbe29fd83f61">

**Comment by getdave** [comment]
> @getdave can I contribute to the `Button` block link editor with `nofollow` control? Thanks <img alt="image" width="640" src="https://user-images.githubusercontent.com/10071857/264918671-443e747f-2867-4513-93ee-dbe29fd83f61.png">

Absolutely! That would be great. You would need to wire everything up to save the nofollow into the block attributes. I look forward to your PR. Please ping me.

**Comment by bangank36** [comment]
@getdave with the Button Link has been merged https://github.com/WordPress/gutenberg/pull/54110, do you think Navigation Links should be next, or do you have other plan for it? Thanks

**Comment by getdave** [comment]
Is there a strong use case for nofollow on Navigation links? How often would one wish to add a link to a Navigation that points to another site? I guess if it were a Navigation other than one in a header of the website perhaps that might happen?

**Comment by bangank36** [comment]
I think user may want to add nofollow to External link Item on the Navgation? Actually there are a handful of articles that help users to add nofollow to their Navigation items, eg: [/how-to-add-nofollow-links-in-wordpress-navigation-menus/](https://www.wpbeginner.com/wp-tutorials/how-to-add-nofollow-links-in-wordpress-navigation-menus/), I guess users will expect the same with the Navigation block

**Comment by skorasaurus** [comment]
can close https://github.com/WordPress/gutenberg/issues/13542 when this is also closed
